ravenframework/Models/ROM.py

```diff
@@ -326,64 +328,97 @@ def train(self,trainingSet):
       @ In, trainingSet, dict or PointSet or HistorySet, data used to train the ROM; if an HistorySet is provided the a list of ROM is created in order to create a temporal-ROM
       @ Out, None
     """
-    if type(trainingSet).__name__ == 'ROM':
-      self.trainingSet              = copy.copy(trainingSet.trainingSet)
-      self.amITrained               = copy.deepcopy(trainingSet.amITrained)
-      self.supervisedContainer      = copy.deepcopy(trainingSet.supervisedContainer)
-      self.seed = trainingSet.seed
+    if isinstance(trainingSet, ROM):
+      self.trainFromInstance(trainingSet)
     else:
       # TODO: The following check may need to be moved to Dummy Class -- wangc 7/30/2018
-      if type(trainingSet).__name__ != 'dict' and trainingSet.type == 'HistorySet':
+      if type(trainingSet) != dict and trainingSet.type == 'HistorySet':
         if not trainingSet.checkIndexAlignment(indexesToCheck=self.pivotParameterId):
-          self.raiseAnError(IOError, "The data provided by the data object", trainingSet.name, "is not synchonized!",
-                  "The time-dependent ROM requires all the histories are synchonized!")
-      self.trainingSet = copy.copy(self._inputToInternal(trainingSet))
-      self._replaceVariablesNamesWithAliasSystem(self.trainingSet, 'inout', False)
+          self.raiseAnError(
+            IOError,
+            f"The data provided by the data object {trainingSet.name}, is not synchonized!",
+            "The time-dependent ROM requires all the histories are synchonized!"
+          )
+
+      # LEGACY SupervisedLearning (SVL) objects train on dictionaries/matrices
+      # New SVL can bypass the data manip and use the dataset directly
+      useDict = self.supervisedContainer[0].needsDictTraining
+      if useDict:
+        self.trainingSet = copy.copy(self._inputToInternal(trainingSet))
+      else:
+        self.trainingSet = trainingSet
 
+      self._replaceVariablesNamesWithAliasSystem(self.trainingSet, 'inout', False)
```

Comment:
Could you check to see if this line works with your proposed data structure? In Model.py, this method only accept dict or list as input.